File: Python/485. Max Consecutive Ones.py
# Time: O(n)
class Solution:
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans, cnt = 0, 0
        for num in nums:
            if num == 1:
                cnt += 1
            else:
                cnt = 0
            ans = max(cnt, ans)
        return ans

File: C++/485. Max Consecutive Ones.cpp
// Time: O(n)
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int ans = 0,sum = 0;
        for(auto num: nums)
        {
            if(num) sum++;
            else sum = 0;
            ans = max(sum,ans);
        }
        return ans;
    }
};